### ISSUE TITLE
Slack alike IRC client

### ISSUE THREAD

**Issue by shime** [mikaelbr/open-source-ideas#14]
## Project description

Slack is a glorification of IRC with search support (for paid plans) and file uploads. Current trend for open source communities is to use Slack as a communication tool, but it all started years ago with IRC. IRC has played a major role in connecting open source communities over the years and is a well known standard in the developer circles.

Slack's main advantage over its competitors is slick design with great color scheme and it would be a good idea to use it as inspiration for this.

There are a lot of IRC clients out there but none of them has great design. The most popular among _real hackers™_ is [irssi](https://irssi.org/) which runs inside a terminal.

The cool part of using Electron for this is that we can reuse the design and logic on the web and create a better experience for web-based IRC clients too, like [Freenode's](https://webchat.freenode.net/).

Drawbacks of using this instead of Slack would be lack of support for search and file uploads.

## Relevant Technology

Electron, React, Redux, Node

## Who is this for

Experienced developers

**Comment by FredrikAugust** [comment]
You also have weechat :) I use that every single day, and absolutely love
it!

**Comment by FredrikAugust** [comment]
Why would you only support MacOS?

**Comment by shime** [comment]
It takes less time to just focus on a single OS for a MVP.

**Comment by FredrikAugust** [comment]
How would people that don't have a mac work on this?

**Comment by Haroenv** [comment]
[Textual](https://www.codeux.com/) already exists though. There are loads of other clients too

**Comment by shime** [comment]
@FredrikAugust I think it's okay to focus on Linux or Mac at first, so choose any you're most comfortable with.

**Comment by FredrikAugust** [comment]
Yeah that sounds better, shouldn't be hard to make it cross-compatible with windows/osx/linux though :)

**Comment by shime** [comment]
@Haroenv looks a bit like Slack, but it's Mac only, closed source and paid, so meh.

**Comment by timwis** [comment]
I highly recommend [irccloud](http://irccloud.com). It's got a modern, slick UI, saves your chat history even when you're offline, and offers nice mobile apps with push notifications.

**Comment by Haroenv** [comment]
Slack is closed source too. I think you're forgetting the value of real employees

**Comment by georoot** [comment]
Has anyone started working on this one ?

**Comment by mikaelbr** [comment]
@georoot As far as I know it's up for grabs. 👍

**Comment by georoot** [comment]
Nice let me give it a shot 👍

**Comment by mikaelbr** [comment]
I've marked it as joined 🎉

**Comment by georoot** [comment]
Hi, just a quick update, if anyone good at css would like to jump in feel free as it is not one of my strongest suite. Here is the repo link https://github.com/georoot/ircc

**Comment by FredrikAugust** [comment]
Seems like @georoot's implementation has staled, so I've started working on this as well.

**Comment by xvilo** [comment]
Cool project. Most IRC clients are just too much to the point. I want a nice interface and more rich media (previews) in my feeds. Would be a great idea. 

You can ultimately create a self-hosted web client for this and also create a hybrid desktop client (same code base) for it. this way you can easily deploy for multiple OS's. 

I'm not a backend dev, so that would be too much for me. But I can help with frontend/design stuff. CSS, Layouts etc.

**Comment by koppor** [comment]
i would build on [Quassel IRC's](https://github.com/quassel/quassel) code, because quassel has the great feature to offer a client/server technology, where the server connects (as IRC client) to the IRC network and the Quassel client is "just" the UI. So, one does not need to reimplement the whole IRC logic, but can focus on the UI.

**Comment by sorki** [comment]
Relevant https://github.com/glowing-bear/glowing-bear

**Comment by benwaffle** [comment]
I recommend contributing to [matrix.org](https://matrix.org) - it already has a large user base, supports file uploads, stickers, end-to-end encryption, is federated, and bridges to other networks including IRC. The client is at [riot.im](https://riot.im/app)

**Comment by sorki** [comment]
@benwaffle while not strictly required it's probably better to run your own matrix bridges (if you care about privacy) which is quite similar to running ZNC to achieve IRC persistence. Both of them require VPS + taking care of the bridge/bouncer which is why people rather go to slack or irccloud (please don't recommend irccloud as it's not open nor free solution).

**Comment by dufferzafar** [comment]
KDE Konversation is an IRC client. It looks pretty standard right now, but they're working on a new UI: https://blogs.kde.org/2017/09/05/konversation-2x-2018-new-user-interface-matrix-support-mobile-version

It'll have matrix support too!

**Comment by elfeffe** [comment]
Like Rocket.chat?

**Comment by testman42** [comment]
Two projects that I wanted to point out that seem relevant to this discussion:
https://thelounge.chat/
https://mattermost.com/

**Comment by aquibbaig** [comment]
What do you make better than Irssi(for terminals)? It provides everything already

**Comment by phun-ky** [comment]
Many people here forget that slack was just IRC with a layer in the beginning :P

**Comment by sygint** [comment]
I love this idea, https://thelounge.chat/ is a pretty nice open source option that has a good interface.

**Comment by KaKi87** [comment]
> slack was just IRC with a layer in the beginning

Just like Reddit's chat is Matrix-based ^^
